### PR TITLE
[456] Feature legacy site news and events on research centre and landing pages

### PR DESCRIPTION
**Pull request by ababic**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
           name: Run Black
           command: |
             . venv/bin/activate
-            black --check ./rca
+            black --check ./rca --exclude migrations/
```
Comment (kevinhowbrook) [inline .circleci/config.yml:63]: yea good idea actually

File: rca/utils/models.py [hunk 2]
```diff
@@ -373,6 +379,84 @@ def get_context(self, request, *args, **kwargs):
...
+        ),
+        through=LegacySiteTaggedPage,
+        blank=True,
+    )
+
+    class Meta:
+        abstract = True
+
+    @property
+    def legacy_news_cache_key(self):
```
Comment (kevinhowbrook) [inline rca/utils/models.py:413]: should this be `legacy_news_and_events_cache_key`? Seen as though it's news _and_ events that will be in the cache. Guess it's only called internally anyway so not a big deal

File: rca/landingpages/models.py [hunk 3]
```diff
@@ -138,7 +144,7 @@ def __str__(self):
         return self.title
 
 
-class LandingPage(BasePage):
+class LandingPage(LegacyNewsAndEventsMixin, BasePage):
```
Comment (kevinhowbrook) [inline rca/landingpages/models.py:147]: I really like the way you've done this, it's going to be really useful down the line for other models

File: rca/utils/models.py (hunk 2 above)
Comment (ababic) [reply rca/utils/models.py:413]: yes, it just seemed to be getting a bit too length/repetitive.

File: rca/landingpages/models.py (hunk 3 above)
Comment (ababic) [reply rca/landingpages/models.py:147]: Cheers! I was tempted to stick this on BasePage, but that would have meant a hell of a lot of migrations :)

**Pull request merged**